python/cucim/tests/unit/clara/test_tiff_read_region.py

```diff
@@ -0,0 +1,157 @@
+#
+# Copyright (c) 2021, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import numpy as np
+import pytest
+
+
+def open_image_cucim(file_path):
+    from cucim import CuImage
+    img = CuImage(file_path)
+    return img
+
+
+def test_tiff_stripe_inner(testimg_tiff_stripe_32x24_16):
+    cucim_img = open_image_cucim(testimg_tiff_stripe_32x24_16)
+    width, height = cucim_img.size('XY')
+    tile_width, tile_height = cucim_img.resolutions['level_tile_sizes'][0]
+
+    # List of ((<start x>, <start y>), (<width>), (<height>))
+    region_list = [((0, 0), (width, height)),  # whole
+                   ((0, 0), (tile_width // 2, tile_height // 2)),  # left-top
+                   ((tile_width // 2, tile_height // 2),
+                    (tile_width, tile_height)),  # middle
+                   ((width - (tile_width // 2), height - (tile_height // 2)),
+                    (tile_width // 2, tile_height // 2)),  # right-bottom
+                   ]
+    for (start_pos, size) in region_list:
+        cucim_arr = np.asarray(cucim_img.read_region(start_pos, size))
+
+        # Not channel values are zero, so we need to check that.
+        channel_value_count = np.apply_along_axis(np.count_nonzero, 2,
+                                                  cucim_arr)
+        count_all_zero = np.count_nonzero(channel_value_count == 0)
+        assert count_all_zero == 0
```

Comment:
```suggestion
        channel_value_count = np.count_nonzero(cucim_arr, axis=2)
        assert np.all(channel_value_count > 0)
```
should be equivalent (avoid `apply_along_axis` which is usually pretty inefficient)